src/components/chats/Chat.tsx

```diff
@@ -0,0 +1,83 @@
+import styled, { css } from "styled-components"
+import { FlexBox } from "components/layout/FlexBox"
+import { TextArea } from "components/forms/textfields/TextArea"
+import { Text } from "components/text/Text"
+import { Button } from "components/forms/Button"
+import { useTheme } from "styles/theme/useTheme"
+
+export type ChatProps = {
+  header?: React.ReactNode
+  chatBody: React.ReactNode
+  footer?: React.ReactNode
+  value?: string
+  onChange: (text: string, e: React.ChangeEvent<HTMLTextAreaElement>) => void
+  onClick: () => void
+}
+
+export const Chat: React.FC<ChatProps> = ({
+  header,
+  chatBody,
+  footer,
+  value,
+  onChange,
+  onClick
+}) => {
+
+  const theme = useTheme()
+
+  return (
+    <Container>
+      {header}
+      {chatBody}
+      {
+        footer
+        ?
+        <Text 
+          _css={css`
+            text-align: center;
+            color: ${theme.styles.textColorSecondary};
+            margin-top: 30px;
+          `}
+        >
+          {footer}
+        </Text>
+        :
+        null
+      }
+      <FlexBox 
+        withRows
+        _css={css`
+          position: absolute;
+          bottom: 20px;
+          left: 0px;
+          width: 100%;
+        `}
+      >
+        <TextArea 
+          onChange={onChange}
+          value={value}
+        />
+        <ButtonContainer>
+          <Button onClick={onClick}>
+            Send
+          </Button>
+        </ButtonContainer>
+      </FlexBox>
+    </Container>
+  )
+}
+
+const Container = styled.div`
+  position: relative;
+  width: 100%;
+  height: 80vh;
```

Comment:
This non-flexible height seems like it can cause future trouble if this component is ever nested in another component that doesn't fill the whole page. Can you have it just fill the parent component?

Reply:
I wasn't sure about how to set the height. Because parent components can have other components on top or below this chat component. The only way I think it should be done is by providing a height props. Could you please propose a solution?

Reply:
Height could be 100%.

If a parent component has a structure of:
```
<div css={`height: 200px;`} />
  <div css={`height: 500px;`} >
    <Chat />
  </div>
<div css={`height: 200px;`} />
```

The chat component would/should take up the 500px space.
Many of the components in the frontend repo use the FlexBox component, which can allow for a bit more flexibility with container sizes auto expanding/contracting.